pepper-apis/dss-core/src/main/java/org/broadinstitute/ddp/Housekeeping.java

```diff
@@ -538,22 +562,87 @@ private static void setupTaskReceiver(Config cfg, String projectId) {
+        ConfigUtil.getString(fileUploadConfig, configTopicNamePath)
+                .map(name -> ProjectTopicName.of(projectId, name))
+                .ifPresentOrElse(topicName -> {
+                    pubsubManager.createTopicIfNotExists(topicName);
+                    storageClient.enableNotifications(uploadsBucketName, topicName, uploadsNotificationsReceiver.acceptedEventTypes());
+
+                    final var subscription = Subscription.newBuilder()
+                            .setName(configSubscriptionName.toString())
+                            .setTopic(topicName.toString())
+                            .build();
+                    pubsubManager.createSubscriptionIfNotExists(subscription);
+                }, () -> {
+                        log.info("No value for key '{}', not creating Pub/Sub topic, subscription, & Cloud Storage notifications",
+                                configTopicNamePath);
+                    });
+
+        // For now, handle messages one at a time so we don't overwhelm resources.
+        final var threadFactory = new ThreadFactoryBuilder()
+                .setNameFormat("uploads-notifications-%d")
+                .setDaemon(true)
+                .build();
+
+        // For now, handle messages one at a time so we don't overwhelm resources.
+        final var executorProvider = InstantiatingExecutorProvider.newBuilder()
+                .setThreadFactory(threadFactory)
+                .build();
+
+        final var subscriberBuilder = Subscriber.newBuilder(configSubscriptionName, uploadsNotificationsReceiver)
+                .setExecutorProvider(executorProvider);
+            
+        startNewSubscriberWithRecovery(configSubscriptionName,
+                subscriberBuilder,
+                executorProvider,
+                MoreExecutors.directExecutor(),
+                subscriber -> {
+                    log.info("Started file uploads subscriber for subscription {}", configSubscriptionName);
+                    uploadsNotificationSubscriber = subscriber;
+                });
+    }
+
+    private static void setupFileScanResultReceiver(Config cfg, GoogleCredentials credentials, String projectId) {
         boolean enabled = cfg.getBoolean(ConfigFile.FileUploads.ENABLE_SCAN_RESULT_HANDLER);
         if (!enabled) {
-            log.warn("File scan result handler is not enabled");
+            log.warn("File scan result handler is disabled ({} == false)", ConfigFile.FileUploads.ENABLE_SCAN_RESULT_HANDLER);
             return;
         }
 
         var subName = ProjectSubscriptionName.of(projectId,
                 cfg.getString(ConfigFile.FileUploads.SCAN_RESULT_SUBSCRIPTION));
 
-        boolean ensureDefault = cfg.getBoolean(ConfigFile.REQUIRE_DEFAULT_GCP_CREDENTIALS);
-        GoogleCredentials credentials = GoogleCredentialUtil.initCredentials(ensureDefault);
-        if (credentials == null) {
-            throw new DDPException("Could not get bucket credentials");
-        }
-
```

Comment:
This was moved to the enclosing method, and the credentials are passed in via a method parameter.

Reply:
No longer applicable